waspc/src/Wasp/Analyzer/ErrorMessage.hs

```diff
@@ -0,0 +1,19 @@
+module Wasp.Analyzer.ErrorMessage
+  ( makeFullErrorMsg,
+  )
+where
+
+import Data.List (intercalate)
+import Wasp.Util (indent)
+
+makeFullErrorMsg :: String -> [String] -> String
+makeFullErrorMsg errorMsg errorCtxMsgs
+  | null errorCtxMsgs = errorMsg
+  | otherwise = intercalate "\n\n" [errorMsg, concatCtxMessages errorCtxMsgs]
```

Comment:
Maaaaybe simpler, but do whatever you like:

```suggestion
makeFullErrorMsg errorMsg errorCtxMsgs = 
  errorMsg ++ (if null errorCtxMsgs then "" else "\n\n" ++ concatCtxMessages errorCtxMsgs)
```

Reply:
I have to disagree with this one - I prefer the variant with guards: less concatenation, seems more declarative (kind of reminds me of a math formula) and easier to edit.

Reply:
Ha ok, no worries! What I liked about the second option is that it reads as "Full error messages consists of error message + something, and that something is nothing if no error ctx messages or if there are some, they are concatenated".
So this part of errorMessage + smth is what I liked.

While in the current code, it is:
1. if no error ctx messages, then it is error message.
2. if error ctx messages, then it is error message + concatenated context error messages.
So in a way, there is duplication, which is this logic of putting error message in front, and then context error messages in the back.

Maybe middle ground is:
```hs
makeFullErrorMsg errorMsg errorCtxMsgs = errorMsg ++ case errorCtxMsgs of
  [] -> ""
  msgs -> "\n\n" ++ concatCtxMessages msgs
```

At the end, this piece of code is so small that all goes, so the whole discussion is a bit trivial :D, but maybe explaining the way of thinking is still interesting.